src/btree/bt_handle.c

```diff
@@ -540,8 +540,13 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt, bool is_ckpt)
     else
         F_CLR(btree, WT_BTREE_NO_CHECKPOINT);
 
-    /* Detect if the btree is disaggregated (for now, just use the file extension). */
-    if (WT_SUFFIX_MATCH(btree->dhandle->name, ".wt_stable")) {
+    /*
+     * Detect if the btree is disaggregated. TODO the file extension check should be replaced with
+     * something more robust.
+     */
```

Comment:
Using an oligarch table implies that the `.wt_stable` file must use disagg block manager right?
Maybe we should set `block_manager=disagg` in the config for the wt_stable when it is created in schema_create?  If that's in place, then "something more robust" in the comment, is merely removing the .wt_stable check, and just checking the `block_manager=` config here.

I think what you have here is fine - I'm just thinking out loud to the next step when the pantry block manager is removed.

Reply:
Yes, that's a good call. I made SLS-527 to track this.